app/lib/middleware/hudApi.js

```diff
@@ -1,80 +1,154 @@
-const fileAsync = require('lowdb/lib/storages/file-async')
+const request = require('request-promise')
+const moment = require('moment')
+const uuid = require('uuid/v4')
+const URL = require('url')
+const fema = require('./fema')
+const oauth2 = require('./oauth2')
+const DRDP_API_BASE = `${process.env.HUD_API_BASE_URL}/drdp/api/v1.0`
+const DRGR_API_BASE = `${process.env.HUD_API_BASE_URL}/drgr/api/v1.0`
+const path = require('path')
+const fs = require('fs')
+const certPath = path.join(__dirname,'..','..','certs','HUD-CA.pem')
+const ca = fs.readFileSync(certPath)
+let oauthTokens = {}
 
-// Start database using file-async storage
-// For ease of use, read is synchronous
-const db = low('mock-data/REAC Modified Database.json', {
-  storage: fileAsync
-})
+const getTokens = () => {
+  return Promise.all(
+  [
+    oauth2.getOAuth2TokenParam('DRGR'),
+    oauth2.getOAuth2TokenParam('DRDP')
+  ])
+}
 
-const databaseFieldReference = {
-  disaster: 'dster_id',
-  state: 'dmge_state_cd',
-  city: 'dmge_city_name',
-  county: 'cnty_name',
-  congrdist: 'fcd_fips91_cd'
+const LOCALE_TYPES = {
+  city: 'cities',
+  county: 'counties',
+  congrdist: 'districts',
+  zipcode: 'zipcodes',
+  township: 'townships',
+  tract: 'tracts'
 }
 
-const decodeField = (fieldname) => {
-  return databaseFieldReference[fieldname]
+const DATABASEFIELDREFERENCE = {
+  disaster: 'DSTER_ID',
+  state: 'DMGE_STATE_CD',
+  city: 'DMGE_CITY_NAME',
+  county: 'CNTY_NAME',
+  congrdist: 'FCD_FIPS_91_CD',
+  zipcode: 'STD_BASC_ZIP_CD',
+  township: 'CNTY_SUB_DIVN_CURR_NAME',
+  tract: 'TRACT_CENSUS_2010_CD'
 }
 
-const getData = function (queryObj, summaryCols, selectCols) {
-  console.log(JSON.stringify(queryObj))
-  console.log(JSON.stringify(summaryCols))
-  console.log(JSON.stringify(selectCols))
-  var result = db.get('disasterRecs').value()
-  for (var phrase in queryObj) {
-    var query = queryObj[phrase]
-    var column = _.keys(query)[0]
-    console.log(`column: ${JSON.stringify(column)}`)
-    var argument = query[column]
-    console.log(`argument: ${JSON.stringify(argument)}`)
-    console.log(`before applying arguments, result length: ${result.length}`)
-    var compositeResult = []
-    _.each(argument, val => {
-      var arg = {}
-      arg[column] = val
-      console.log(`applying arg: ${JSON.stringify(arg)}`)
-      compositeResult = _.concat(compositeResult, _.filter(result, arg))
-      console.log(`compositeResult length after applying: ${compositeResult.length}`)
-    })
-    result = compositeResult
-    console.log(`compositeResult length: ${compositeResult.length}`)
-    console.log(`result length: ${result.length}`)
-  }
-  if (summaryCols) return summarizeCols(result, summaryCols)
-  if (selectCols) {
-    result = _.map(result, rec => {
-      var retValue = {}
-      _.forEach(selectCols, col => { retValue[col] = rec[col] })
-      return retValue
+const SERVICE_CONSUMER_DATA = {
+  'auditCorrelationId': '17f24136-2494-4bf8-9d3b-9baafaae0cc9',
```

Comment:
What is this?

Reply:
That's information that is required by the MULE service.  Willie specified that we needed to pass that info.

Reply:
Interesting!